File: tests/variables.py
def assignment(a: bool, b: str, c: int, d: int) -> int:
    e = a
    f = b
    g = c
    h = c + d
    j = k = l = c + d + g + h
    return j + k + l if a and e or f == "hello" else 0

def annotated_assignment(a: bool, b: str, c: int, d: int) -> int:
    e: int = 3
    f: bool = a
    g: str = b
    h: int = c + d
    return e + h if a and f and g == "hello" else 0

def aug_assign(a: int, b: int) -> int:
    c = 0
    c += a
    c -= b
    c /= (a + b)
    c //= a
    c *= b
    return c

def repeated_assign(a: str, b: str) -> str:
    c = a
    c = b
    if b == a:
        c = "equal"
    return c

def pathological(a: bool, b: str, c: str) -> str:
    if a:
        d = b
    else:
        d = c
    return d

File: temp/variables.rs
pub fn assignment(a: bool, b: &str, c: i64, d: i64) -> i64 {
    let e = a;
    let f = b;
    let g = c;
    let h = c + d;
    let j = c + d + g + h;
    let k = j;
    let l = j;
    return if a && e || f == "hello" { j + k + l } else { 0 };
}

pub fn annotated_assignment(a: bool, b: &str, c: i64, d: i64) -> i64 {
    let e: i64 = 3;
    let f: bool = a;
    let g: String = b;
    let h: i64 = c + d;
    return if a && f && g == "hello" { e + h } else { 0 };
}

pub fn aug_assign(a: i64, b: i64) -> i64 {
    let mut c = 0;
    c += a;
    c -= b;
    c /= a + b;
    c /= a;
    c *= b;
    return c;
}

pub fn repeated_assign(a: &str, b: &str) -> String {
    let mut c = a;
    c = b;
    if b == a {
        c = "equal";
    }
    return c;
}

pub fn pathological(a: bool, b: &str, c: &str) -> String {
    let mut d: &str = "";
    if a {
        d = b;
    } else {
        d = c;
    }
    return d;
}

